Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -922,72 +920,61 @@ bool WebGLRenderingContextBase::clearIfComposited(WebGLRenderingContextBase::Cal
     // `clearIfComposited()` is a function that prepares for updates. Mark the context as active.
     updateActiveOrdinal();
 
-    GCGLbitfield buffersNeedingClearing = m_context->getBuffersToAutoClear();
+    GCGLbitfield dirtyBuffersMask = m_defaultFramebuffer->dirtyBuffers();
 
-    if (!buffersNeedingClearing || (mask && m_framebufferBinding) || (m_rasterizerDiscardEnabled && caller == CallerTypeDrawOrClear))
+    if (!dirtyBuffersMask || (mask && m_framebufferBinding) || (m_rasterizerDiscardEnabled && caller == CallerTypeDrawOrClear))
         return false;
 
-    // Use the underlying GraphicsContext3D's attributes to take into
-    // account (for example) packed depth/stencil buffers.
-    auto contextAttributes = m_context->contextAttributes();
-
     // Determine if it's possible to combine the clear the user asked for and this clear.
     bool combinedClear = mask && !m_scissorEnabled;
 
-    m_context->disable(GraphicsContextGL::SCISSOR_TEST);
-    if (combinedClear && (mask & GraphicsContextGL::COLOR_BUFFER_BIT) && (m_backDrawBuffer != GraphicsContextGL::NONE)) {
-        m_context->clearColor(m_colorMask[0] ? m_clearColor[0] : 0,
-                              m_colorMask[1] ? m_clearColor[1] : 0,
-                              m_colorMask[2] ? m_clearColor[2] : 0,
-                              m_colorMask[3] ? m_clearColor[3] : 0);
-    } else
-        m_context->clearColor(0, 0, 0, 0);
-    if (m_oesDrawBuffersIndexed)
-        m_context->colorMaskiOES(0, true, true, true, true);
-    else
-        m_context->colorMask(true, true, true, true);
-    GCGLbitfield clearMask = GraphicsContextGL::COLOR_BUFFER_BIT;
-    if (contextAttributes.depth) {
+    if (m_scissorEnabled)
+        m_context->disable(GraphicsContextGL::SCISSOR_TEST);
```

Comment:
Should this be an RAII helper?

Reply:
Yep. I'll do them in a separate commti.

Reply:
In fact, I had an amend that I hadn't uploaded.. :(